riverdriver/riversqlite/internal/dbsqlc/river_job.sql

```diff
@@ -0,0 +1,426 @@
+    @kind,
+    @max_attempts,
+    json(cast(@metadata AS blob)),
+    @priority,
+    @queue,
+    coalesce(cast(sqlc.narg('scheduled_at') AS text), datetime('now', 'subsec')),
+    @state,
+    json(cast(@tags AS blob)),
+    CASE WHEN length(cast(@unique_key AS blob)) = 0 THEN NULL ELSE @unique_key END,
+    @unique_states
+)
+ON CONFLICT (unique_key)
+    WHERE unique_key IS NOT NULL
+        AND unique_states IS NOT NULL
+        AND CASE state
+                WHEN 'available' THEN unique_states & (1 << 0)
+                WHEN 'cancelled' THEN unique_states & (1 << 1)
+                WHEN 'completed' THEN unique_states & (1 << 2)
+                WHEN 'discarded' THEN unique_states & (1 << 3)
+                WHEN 'pending'   THEN unique_states & (1 << 4)
+                WHEN 'retryable' THEN unique_states & (1 << 5)
+                WHEN 'running'   THEN unique_states & (1 << 6)
+                WHEN 'scheduled' THEN unique_states & (1 << 7)
+                ELSE 0
+            END >= 1
+    -- Something needs to be updated for a row to be returned on a conflict.
+    DO UPDATE SET kind = EXCLUDED.kind;
+
+-- name: JobInsertFull :one
+INSERT INTO /* TEMPLATE: schema */river_job(
+    args,
+    attempt,
+    attempted_at,
+    attempted_by,
+    created_at,
+    errors,
+    finalized_at,
+    kind,
+    max_attempts,
+    metadata,
+    priority,
+    queue,
+    scheduled_at,
+    state,
+    tags,
+    unique_key,
+    unique_states
+) VALUES (
+    @args,
+    @attempt,
+    cast(sqlc.narg('attempted_at') as text),
+    CASE WHEN length(cast(@attempted_by AS blob)) = 0 THEN NULL ELSE json(@attempted_by) END,
+    coalesce(cast(sqlc.narg('created_at') AS text), datetime('now', 'subsec')),
+    CASE WHEN length(cast(@errors AS blob)) = 0 THEN NULL ELSE @errors END,
+    cast(sqlc.narg('finalized_at') as text),
+    @kind,
+    @max_attempts,
+    json(cast(@metadata AS blob)),
+    @priority,
+    @queue,
+    coalesce(cast(sqlc.narg('scheduled_at') AS text), datetime('now', 'subsec')),
+    @state,
+    json(cast(@tags AS blob)),
+    CASE WHEN length(cast(@unique_key AS blob)) = 0 THEN NULL ELSE @unique_key END,
+    @unique_states
+) RETURNING *;
+
+-- name: JobList :many
+SELECT *
+FROM /* TEMPLATE: schema */river_job
+WHERE /* TEMPLATE_BEGIN: where_clause */ 1 /* TEMPLATE_END */
+ORDER BY /* TEMPLATE_BEGIN: order_by_clause */ id /* TEMPLATE_END */
+LIMIT @max;
+
+-- Rescue a job.
+--
+-- This is supposed to be a batch insert, but various limitations of the
+-- combined SQLite + sqlc has left me unable to find a way of injecting many
+-- arguments en masse (like how we slightly abuse arrays to pull it off for the
+-- Postgres drivers), so we loop over many insert operations instead, with the
+-- expectation that this may be fixable in the future. Because SQLite targets
+-- will often be local and therefore with a very minimal round trip compared to
+-- a network, looping over operations is probably okay performance-wise.
```

Comment:
Comment not applicable to this query?

Reply:
I had copied it from another query, but it still is somewhat relevant in notes on what makes it different from the Postgres `JobRescue` implementation. I just tailored it a bit better to this specific query.